static/app/components/sidebar/broadcasts.tsx

```diff
@@ -10,194 +8,112 @@ import SidebarPanel from 'sentry/components/sidebar/sidebarPanel';
-      hidePanel={hidePanel}
-    >
-      {loading ? (
-        <LoadingIndicator />
-      ) : broadcasts.length === 0 ? (
-        <SidebarPanelEmpty>
-          {t('No recent updates from the Sentry team.')}
-        </SidebarPanelEmpty>
-      ) : (
-        broadcasts.map(item => (
-          <BroadcastPanelItem
-            key={item.id}
-            hasSeen={item.hasSeen}
-            title={item.title}
-            message={item.message}
-            link={item.link}
-            mediaUrl={item.mediaUrl}
-            category={item.category}
-          />
-        ))
-      )}
-    </SidebarPanel>
-  );
-}
-
-class Broadcasts extends Component<Props, State> {
-  state: State = {
-    broadcasts: [],
-    loading: true,
-  };
-
-  componentDidMount() {
-    this.fetchData();
-    document.addEventListener('visibilitychange', this.handleVisibilityChange);
-  }
-
-  componentWillUnmount() {
-    window.clearTimeout(this.markSeenTimeout);
-    window.clearTimeout(this.pollingTimeout);
-
-    document.removeEventListener('visibilitychange', this.handleVisibilityChange);
-  }
-
-  pollingTimeout: number | undefined = undefined;
-  markSeenTimeout: number | undefined = undefined;
-
-  startPolling() {
-    if (this.pollingTimeout) {
-      this.stopPolling();
+  onShowPanel,
+}: CommonSidebarProps) {
+  const api = useApi();
+  const organization = useOrganization();
+  const [hasSeenAllPosts, setHasSeenAllPosts] = useState(false);
+  const previousPanel = usePrevious(currentPanel);
+
+  const {isPending, data: broadcasts = []} = useApiQuery<Broadcast[]>(
+    [`/organizations/${organization.slug}/broadcasts/`],
+    {
+      staleTime: 0,
+      refetchInterval: POLLER_DELAY,
     }
-    this.pollingTimeout = window.setTimeout(this.fetchData, POLLER_DELAY);
-  }
-
-  stopPolling() {
-    window.clearTimeout(this.pollingTimeout);
-    this.pollingTimeout = undefined;
-  }
+  );
 
-  fetchData = async () => {
-    if (this.pollingTimeout) {
-      this.stopPolling();
-    }
+  const {mutate: markBroadcastsAsSeen} = useMutation({
+    mutationFn: (unseenPostIds: string[]) => {
+      return api.requestPromise('/broadcasts/', {
+        method: 'PUT',
+        query: {id: unseenPostIds},
+        data: {hasSeen: '1'},
+      });
+    },
+  });
+
+  const unseenPostIds = useMemo(
+    () => broadcasts.filter(item => !item.hasSeen).map(item => item.id),
+    [broadcasts]
+  );
 
-    try {
-      const data = await getAllBroadcasts(this.props.api, this.props.organization.slug);
-      this.setState({loading: false, broadcasts: data || []});
-    } catch {
-      this.setState({loading: false});
+  const handleShowPanel = useCallback(() => {
+    if (unseenPostIds.length) {
+      markBroadcastsAsSeen(unseenPostIds);
     }
 
-    this.startPolling();
-  };
-
-  /**
-   * If tab/window loses visibility (note: this is different than focus), stop
-   * polling for broadcasts data, otherwise, if it gains visibility, start
-   * polling again.
-   */
```

Comment:
by default React Query does not automatically refetch when a tab or window loses visibility

Reply:
we have that turned off in useApiQuery, you might have to explicitly re-enable it.